cl/citations/tests.py

```diff
@@ -2161,3 +2195,54 @@ async def test_can_extract_all_citations_within_text(self) -> None:
 
         clusters = second_citation["clusters"]
         self.assertEqual(len(clusters), 0)
+
+    @override_settings(MAX_CITATIONS_PER_REQUEST=10)
+    async def test_can_limit_max_citations_per_request(
+        self, cache_key_mock
+    ) -> None:
+        ten_citations = "56 F.2d 9, " * 10
+        text_citation = f"{ten_citations} 139 U.S. 601, 155 U.S. 597"
+        r = await self.async_client.post(
+            reverse("citation-lookup-list", kwargs={"version": "v3"}),
+            {"text": text_citation},
+        )
+
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        data = json.loads(r.content)
+        self.assertEqual(len(data), 11)
+
+        last_citation = data[-1]
+        self.assertEqual(last_citation["citation"], "139 U.S. 601")
+        self.assertEqual(last_citation["status"], HTTPStatus.TOO_MANY_REQUESTS)
+        self.assertEqual(last_citation["error_message"], "Too many requests.")
+
+    @patch(
+        "cl.api.utils.CitationCountRateThrottle.get_citations_rate",
+        return_value=(20, 60),
```

Comment:
The `get_citations_rate` method currently returns a `Tuple[int, int]`. This is because it parses the citation rate before returning the value. We can refactor this code to return the rate as a `string`, making it consistent with the other Throttle classes in DRF. The [get_rate](https://github.com/encode/django-rest-framework/blob/9864c47018238dd9f8477df0e4ab2ef62a24fd2b/rest_framework/throttling.py#L82-L95) method in the SimpleRateThrottle class returns the rate as a string the includes the value and the unit( e.g., "20/s", "60/m", "250/h")

Reply:
Hm, I'm even more confused now, because looking at `get_citations_rate`, it looks like it either does:

```
default_rate = self.THROTTLE_RATES["citations"]
```
Which returns a `str` like `"60/min"`, or it gets overridden, which does:

```
        custom_rate = settings.REST_FRAMEWORK[
            "CITATION_LOOKUP_OVERRIDE_THROTTLE_RATES"
        ].get(request.user.username, None)
```

In that case, it returns something like, say, `"1000/min"`, no? I don't see where it returns either a Tuple or anything other than a string?

Reply:
> I don't see where it returns either a Tuple or anything other than a string.

I recently updated this helper function in commit [c1858a0](https://github.com/freelawproject/courtlistener/pull/3943/commits/c1858a075d39be5e6e078949b8243176aa340539). This change simplifies the function's output by returning a single string value instead of a tuple. Previously, the helper function returned the value from calling `self.parse_rate(custom_rate or default_rate)` and self.parse_rate() returns tuples.

Reply:
Oh, whew, I'm not crazy. Good.

Reply:
> I'm not crazy.

No, you're not 😄. It's my fault. I should've added a comment saying I updated the code because your initial comment made me realize that returning strings makes a lot more sense for this particular helper function.